File: binary_search.py
from random import randint
from insertion_sort import *

# Returns position in array if found otherwise returns -1
def binary_search(arr, target):
	bottom = 0
	top = len(arr) - 1

	while bottom <= top:
		mid = (top + bottom) // 2

		if arr[mid] == target:
			return mid
		elif arr[mid] > target:
			top = mid - 1
		else:
			bottom = mid + 1

	return -1

if __name__ == "__main__":
	arr = []
	target = randint(1, 20)
	
	for i in range(10):
		arr.append(randint(1, 20))

	arr = insertion_sort(arr)

	print(f"Array: {arr}")
	print(f"Target: {target}")
	print(f"Position found: {binary_search(arr, target)}")

File: js/binary_search.js
function binarySearch(arr, target) {
	let top = arr.length - 1;
	let bottom = 0;
	let mid;

	while (bottom <= top) {
		mid = Math.floor((top + bottom) / 2);

		if (arr[mid] === target) return mid;
		else if (arr[mid] > target) top = mid - 1;
		else bottom = mid + 1;
	}

	return -1;
}

const arr = [1, 2, 3, 4, 5, 6];
const target1 = 5;
const target2 = 9;

console.log(binarySearch(arr, target1));
console.log(binarySearch(arr, target2));